azext_edge/edge/providers/orchestration/upgrade.py

```diff
@@ -248,17 +262,21 @@ def _stop_display(self):
 
     def _process(self):
         if self.require_instance_upgrade:
+
+            # prep the instance
+            self.instance.pop("systemData", None)
+            inst_props = self.instance["properties"]
+            if inst_props.get("schemaRegistryRef", {}).get("resourceId"):
+                self.sr_resource_id = inst_props["schemaRegistryRef"]["resourceId"]
```

Comment:
Lookup doesn't have to happen twice, it can be simplified like:

```python
            self.sr_resource_id = inst_props.get("schemaRegistryRef", {}).get("resourceId")
            if not self.sr_resource_id:
                raise RequiredArgumentMissingError(
                    "Cannot determine the schema registry id from installed extensions, please provide the schema "
                    "registry id via `--sr-id`."
                )
            inst_props["schemaRegistryRef"] = {"resourceId": self.sr_resource_id}
```

Reply:
smart